pandas/core/generic.py

```diff
@@ -1192,125 +1205,94 @@ def _slice(self, slobj, axis=0, kind=None):
 
         """
         axis = self._get_block_manager_axis(axis)
-        result = self._constructor(self._data.get_slice(slobj, axis=axis))
-        result = result.__finalize__(self)
-
-        # this could be a view
-        # but only in a single-dtyped view slicable case
-        is_copy = axis!=0 or result._is_view
-        result._set_is_copy(self, copy=is_copy)
-        return result
+        data = self._data.get_slice(slobj, axis=axis)
+        return self._constructor(data)._set_parent(self).__finalize__(self)
 
     def _set_item(self, key, value):
+
+        self._check_copy_on_write()
         self._data.set(key, value)
         self._clear_item_cache()
 
-    def _set_is_copy(self, ref=None, copy=True):
-        if not copy:
-            self.is_copy = None
-        else:
-            if ref is not None:
-                self.is_copy = weakref.ref(ref)
-            else:
-                self.is_copy = None
-
-    def _check_is_chained_assignment_possible(self):
-        """
-        check if we are a view, have a cacher, and are of mixed type
-        if so, then force a setitem_copy check
-
-        should be called just near setting a value
+    def _set_parent(self, ref=None, copy=True):
+        if ref is not None:
+            self._parent.extend(ref._parent)
+            self._parent.append(weakref.ref(ref))
+        return self
 
-        will return a boolean if it we are a view and are cached, but a single-dtype
-        meaning that the cacher should be updated following setting
-        """
-        if self._is_view and self._is_cached:
-            ref = self._get_cacher()
-            if ref is not None and ref._is_mixed_type:
-                self._check_setitem_copy(stacklevel=4, t='referant', force=True)
-            return True
-        elif self.is_copy:
-            self._check_setitem_copy(stacklevel=4, t='referant')
-        return False
+    def _check_copy_on_write(self):
 
-    def _check_setitem_copy(self, stacklevel=4, t='setting', force=False):
-        """
+        # we could have a copy-on-write scenario
+        if self._parent and self._parent_copy_on_write:
 
-        Parameters
-        ----------
-        stacklevel : integer, default 4
-           the level to show of the stack when the error is output
-        t : string, the type of setting error
-        force : boolean, default False
-           if True, then force showing an error
+            # we have an exception
+            if isinstance(self._parent, Exception):
```

Comment:
hah, I need this for defered evaluation (e.g. see the ``.dt.hour = 5`` example above). I agree its hacky, (i mean I could define a different kind of 'parent' that would work similarly, but this is just easy).